File: GoF/comportamentais/template_method/python/csv_dataminer.py
from dataminer import DataMiner


class CsvDataMiner(DataMiner):
    def open_file(self, path):
        print("Arquivo CSV aberto")
        return "Arquivo CSV"

    def extract_data(self, file):
        return "Dados do arquivo CSV"

    def parse_data(self, raw_data):
        return raw_data + " | Dados CSV tratados"

    def close_file(self, file):
        print("Arquivo CSV fechado")

File: GoF/comportamentais/template_method/java/CsvDataMiner.java
public class CsvDataMiner extends DataMiner{
    @Override
    public String openFile(String path){
        System.out.println("Arquivo CSV aberto");
        return "Arquivo CSV";
    }

    @Override
    public String extractData(String file){
        return "Dados do arquivo CSV";
    }

    @Override
    public String parseData(String rawData){
        return rawData + " | Dados CSV tratados";
    }

    @Override
    public void closeFile(String file){
        System.out.println("Arquivo CSV fechado");
    }

}